server/routes/user.js

```diff
@@ -0,0 +1,85 @@
+const express = require("express");
+const router = express.Router();
+const { check, validationResult } = require("express-validator");
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+
+//Models
+const User = require("../models/User");
+
+//MiddleWare
+const auth = require("../middleWare/auth");
+
+//@route /user/signup
+//@desc signin route
+//@route public
+router.post(
+  "/signup",
+  [
+    check("email", "please include a valid email").isEmail(),
+    check("password", "please enter a password with 6").isLength({
+      min: 6
+    })
+  ],
+  async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      //return error in json format
+      return res.status(400).json({ errors: errors.array() });
+    }
+    const { email, password } = req.body;
+    try {
+      // See if user exists
+      let user = await User.findOne({ email });
+
+      if (user) {
+        //return error in json format
+        res.status(400).json({ errors: [{ msg: "User already exists" }] });
+      }
+
+      user = new User({
+        email,
+        password
+      });
+      // Encrypt password
+      const salt = await bcrypt.genSalt(10);
+      user.password = await bcrypt.hash(password, salt);
+
+      // register hashed password and email. it returns ID.
+      await user.save();
+      const payload = {
```

Comment:
userSchema.pre('save', 

i think you should use this kind of function and add all logic in there.